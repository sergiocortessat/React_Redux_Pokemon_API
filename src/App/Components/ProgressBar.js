import PropTypes from 'prop-types';

const ProgressBar = (props) => {
  const { bgcolor, completed } = props;

  const containerStyles = {
    height: 20,
    width: '100%',
    backgroundColor: '#e0e0de',
    borderRadius: 50,
    margin: 10,
  };

  const fillerStyles = {
    height: '100%',
    width: `${completed}%`,
    backgroundColor: bgcolor,
    borderRadius: 'inherit',
    textAlign: 'right',
    transition: 'width 1s ease-in-out',

  };

  const labelStyles = {
    padding: 5,
    color: 'white',
    fontWeight: 'bold',
  };

  return (
    <div style={containerStyles}>
      <div style={fillerStyles}>
        <span style={labelStyles}>{`${completed}%`}</span>
      </div>
    </div>
  );
};
// Proptypes
ProgressBar.propTypes = {
  bgcolor: PropTypes.string.isRequired,
  completed: PropTypes.number.isRequired,
};

export default ProgressBar;
