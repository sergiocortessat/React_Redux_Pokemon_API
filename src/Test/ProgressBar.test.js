import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ProgressBar from '../App/Components/ProgressBar';

describe('Header with Link', () => {
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <ProgressBar />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
