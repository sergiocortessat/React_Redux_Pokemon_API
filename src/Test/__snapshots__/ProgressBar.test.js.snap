// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header with Link should match with snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#e0e0de",
      "borderRadius": 50,
      "height": 20,
      "margin": 10,
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#e0e0de",
        "borderRadius": "inherit",
        "height": "100%",
        "textAlign": "right",
        "transition": "width 1s ease-in-out",
        "width": "0%",
      }
    }
  >
    <span
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
          "padding": 5,
        }
      }
    >
      0 undefined
    </span>
  </div>
</div>
`;
